pkg/kv/kvserver/kvstorage/init.go

```diff
@@ -347,47 +396,103 @@ func LoadAndReconcileReplicas(ctx context.Context, eng storage.Engine) (*EngineR
 					ident.StoreID, desc)
 			}
```

Comment:
I think we could push the invariant check down where you check other invariants. This would allow removing `Replica.descReplicaID` and `setDescReplicaID`.